hasktorch/test/Torch/Typed/FunctionalSpec.hs

```diff
@@ -476,12 +476,35 @@ instance ( TensorOptions shape   dtype device
   apply SymeigSpec _ _ = do
     t <- rand @shape @dtype @device
     foldMap
-      (\(eigenvectors, upper) -> do
-        let (t', t'') = symeig eigenvectors upper t
+      (\upper -> do
+        let (t', t'') = symeig upper t
         checkDynamicTensorAttributes t'
         checkDynamicTensorAttributes t''
       )
-      ((,) <$> [True, False] <*> [D.Upper, D.Lower])
+      [D.Upper, D.Lower]
+
+data SymeigvaluesSpec = SymeigvaluesSpec
+
+instance ( TensorOptions shape   dtype device
+         , TensorOptions shape'  dtype device
+         , TensorOptions shape'' dtype device
+         , shape' ~ VectorOfSquare shape
+         , SymeigDTypeIsValid device dtype
+         , RandDTypeIsValid device dtype
+         )
+  => Apply
+       SymeigvaluesSpec
+       (Proxy device, (Proxy dtype, Proxy shape))
+       (() -> IO ())
+ where
+  apply SymeigvaluesSpec _ _ = do
+    t <- rand @shape @dtype @device
+    foldMap
+      (\upper -> do
+        let t' = symeigvalues upper t
+        checkDynamicTensorAttributes t'
+      )
+      [D.Upper, D.Lower]
```

Comment:
looks good, thanks for making this change!

However, adding an `Apply` instance for this test isn't enough to run it. You need to add a block down in the `spec'` function as well so that the test is run.

I have one more suggestion: Since the constraints, input, and output types for the `Apply` instances for `SymeigSpec` and `SymeigvaluesSpec` are the same, you can merge them:
```haskell
data SymeigSpec = SymeigSpec | SymeigvaluesSpec

instance ( TensorOptions shape   dtype device
         , TensorOptions shape'  dtype device
         , TensorOptions shape'' dtype device
         , shape' ~ VectorOfSquare shape
         , shape'' ~ Square shape
         , SymeigDTypeIsValid device dtype
         , RandDTypeIsValid device dtype
         )
  => Apply
       SymeigSpec
       (Proxy device, (Proxy dtype, Proxy shape))
       (() -> IO ())
 where
   apply SymeigSpec _ _ = ...
   apply SymeigvaluesSpec _ _ = ...
```

Reply:
Thank you for your advice.
I have merged these.